### ISSUE TITLE
why my 'workers=4,use_multiprocessing=True' do not work while training?

### ISSUE THREAD

**Issue by renguote** [keras-team/tf-keras#405]
You can obtain the TensorFlow version with:
TF  2.10
**Describe the problem**.
**Describe the current behavior**.
history = model.fit(train_model_input, epochs=1, verbose=1, validation_split=0.0,steps_per_epoch= math.ceil( train_row_len/256), workers=4,use_multiprocessing=True).
As you see, 'workers=4,use_multiprocessing=True'  don't  work while training.I find it only 100% CPU-Util not the expect 400%.
**Describe the expected behavior**.
I can  increase the utilization of my CPU to 400%.

**Standalone code to reproduce the issue**.

import tensorflow as tf
from tensorflow.keras import datasets, layers, models
from tensorflow.keras.utils import Sequence
from math import ceil
import numpy as np
import numpy as np
import math
import os

class DataGenerator(Sequence):
 
    def __init__(self, x_set, y_set, batch_size):
        self.x, self.y = x_set, y_set
        self.batch_size = batch_size

    def __len__(self):
        return ceil(len(self.x) / self.batch_size)

    def __getitem__(self, idx):
        end = min(self.x.shape[0], (idx + 1)*batch_size)
        return self.x[idx*batch_size:end], self.y[idx*batch_size:end]

batch_size = 32
epochs = 100

(train_images, train_labels), (test_images, test_labels) = datasets.cifar10.load_data()

# Normalize pixel values to be between 0 and 1
train_images, test_images = train_images / 255.0, test_images / 255.0
training_generator = DataGenerator(train_images, train_labels, batch_size)

model = models.Sequential()
model.add(layers.Conv2D(320, (3, 3), activation='relu', input_shape=(32, 32, 3)))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(64, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(64, (3, 3), activation='relu'))
model.add(layers.Flatten())
model.add(layers.Dense(64, activation='relu'))
model.add(layers.Dense(10))

model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])

history = model.fit(x=training_generator,
                    steps_per_epoch=train_images.shape[0]//batch_size,
                    validation_data=(train_images, train_labels),
                    epochs=epochs,
                    verbose=1,
                    use_multiprocessing=True,
                    workers=4)

**Source code / logs**.

**Comment by tilakrayal** [comment]
@sada-narayanappa,
Could you please confirm whether you are facing the issue in **tf-keras** or the **keras3.0**. The related issue was closed and If you need any further assistance, Could you please raise the new request with the reproducible code on [keras-team/keras](https://github.com/keras-team/keras/issues/) repo for keras3.0 (or) [keras-team/tf-keras repo](https://github.com/keras-team/tf-keras/issues) for tf-keras. Thank you!
